doc/source/tune/examples/tune_analyze_results.ipynb

```diff
@@ -0,0 +1,912 @@
+    "3. Basic *trial-level* analysis: access individual trial hyperparameter configs and last reported metrics\n",
+    "4. Plotting the entire history of reported metrics for a trial\n",
+    "5. Accessing saved checkpoints (assuming that you have enabled checkpointing) and loading into a model for test inference\n",
+    "\n",
+    "```python\n",
+    "result_grid: ResultGrid = tuner.fit()\n",
+    "best_result: Result = result_grid.get_best_result()\n",
+    "```\n",
+    "\n",
+    "The output of `tuner.fit()` is a [`ResultGrid`](result-grid-docstring), which is a collection of [`Result`](result-docstring) objects. See the linked documentation references for [`ResultGrid`](result-grid-docstring) and [`Result`](result-docstring) for more details on what attributes are available.\n",
+    "\n",
+    "Let's start by performing a hyperparameter search with the MNIST PyTorch example. The training function is defined {doc}`here </tune/examples/includes/mnist_pytorch>`, and we pass it into a `Tuner` to start running the trials in parallel."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "032e9376-4283-4876-8103-79a9dc3b9b3d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from ray import tune, air\n",
+    "from ray.tune.examples.mnist_pytorch import train_mnist\n",
+    "from ray.tune import ResultGrid\n",
+    "\n",
+    "exp_name = \"tune_analyzing_results\"\n",
+    "tuner = tune.Tuner(\n",
+    "    train_mnist,\n",
+    "    param_space={\n",
+    "        \"lr\": tune.loguniform(0.001, 0.1),\n",
+    "        \"momentum\": tune.grid_search([0.8, 0.9, 0.99]),\n",
+    "        \"should_checkpoint\": True,\n",
+    "    },\n",
+    "    run_config=air.RunConfig(\n",
+    "        name=exp_name,\n",
+    "        stop={\"training_iteration\": 100},\n",
+    "        checkpoint_config=air.CheckpointConfig(\n",
+    "            checkpoint_score_attribute=\"mean_accuracy\",\n",
+    "            num_to_keep=5,\n",
+    "        )\n",
+    "    ),\n",
+    "    tune_config=tune.TuneConfig(mode=\"max\", metric=\"mean_accuracy\", num_samples=3),\n",
+    ")\n",
+    "result_grid: ResultGrid = tuner.fit()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5027136e-94e1-47b7-a12b-94406deb0c14",
+   "metadata": {},
+   "source": [
+    "## Loading experiment results from an directory\n",
+    "\n",
+    "Although we have the `result_grid` object in memory because we just ran the Tune experiment above, we might be performing this analysis after our initial training script has exited. We can retrieve the `ResultGrid` from a [restored `Tuner`](tune-stopping-guide), passing in the experiment directory, which should look something like `~/ray_results/{exp_name}`. If you don't specify an experiment `name` in the `RunConfig`, the experiment name will be auto-generated and can be found in the logs of your experiment.\n",
+    "\n",
+    "```python\n",
```

Comment:
could we just have this as an actual notebook cell?

Reply:
I believe all the ipynb files are already tested in the CI. This cell errored in the CI, since it uses some temp directory and is not at `~/ray_results`. I can change the `local_dir` for this example to try and fix this.